src/main/java/fi/aalto/cs/apluscourses/intellij/model/IntelliJLibrary.java

```diff
@@ -42,8 +42,13 @@ protected void loadInternal() {
     for (String uri : getUris()) {
       library.addRoot(uri, OrderRootType.CLASSES);
     }
+
     //HACK: this is the only way to access properties that I am aware of
-    initializeLibraryProperties(((LibraryEx) library).getProperties());
+    LibraryEx.ModifiableModelEx libraryEx = (LibraryEx.ModifiableModelEx) library;
+    LibraryProperties<S> properties = libraryEx.getProperties();
+    initializeLibraryProperties(properties);
+    libraryEx.setProperties(properties);
```

Comment:
Just to make sure, does this still keep working for 2020.2?

Reply:
Apparently, this code works also in 2020.2.

But because of the changes in **build.gradle**, the plug-in, as such, is not compatible with IDEA 2020.2, and cannot be installed to it. So once we release this, users must update their IntelliJ to 2020.3 to get this update.

But as I said, it seems that this piece of code works also in the older version of IntelliJ. Recap of what this change is about: Basically, what was changed here, is the addition of line 50; previously properties were got with `getProperties()` and modified inside `initializeLibraryProperties()` (just like in the new code), but they were never set back with `setProperties()`. In the old version, it however worked because `getProperties()` returned the reference to the same `LibraryProperties` object that the object itself use (so, calling `setProperties()` changed nothing). In the new version, `getProperties()` returns a copy of the properties object, so the modifications have no effect unless `setProperties()` is called.